Mathlib/Analysis/Asymptotics/Lemmas.lean

```diff
@@ -0,0 +1,792 @@
+/-
+Copyright (c) 2019 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Yury Kudryashov
+-/
+import Mathlib.Analysis.Asymptotics.Defs
+import Mathlib.Analysis.Normed.Group.Bounded
+import Mathlib.Analysis.Normed.Group.InfiniteSum
+import Mathlib.Analysis.Normed.MulAction
+import Mathlib.Topology.Algebra.Order.LiminfLimsup
+import Mathlib.Topology.PartialHomeomorph
+
+/-!
+# Further basic lemmas about asymptotics
+
+-/
+
+open Set Topology Filter NNReal
+
+namespace Asymptotics
+
+
+variable {α : Type*} {β : Type*} {E : Type*} {F : Type*} {G : Type*} {E' : Type*}
+  {F' : Type*} {G' : Type*} {E'' : Type*} {F'' : Type*} {G'' : Type*} {E''' : Type*}
+  {R : Type*} {R' : Type*} {𝕜 : Type*} {𝕜' : Type*}
+
+variable [Norm E] [Norm F] [Norm G]
+variable [SeminormedAddCommGroup E'] [SeminormedAddCommGroup F'] [SeminormedAddCommGroup G']
+  [NormedAddCommGroup E''] [NormedAddCommGroup F''] [NormedAddCommGroup G''] [SeminormedRing R]
+  [SeminormedAddGroup E''']
+  [SeminormedRing R']
+
+variable [NormedDivisionRing 𝕜] [NormedDivisionRing 𝕜']
+variable {c c' c₁ c₂ : ℝ} {f : α → E} {g : α → F} {k : α → G}
+variable {f' : α → E'} {g' : α → F'} {k' : α → G'}
+variable {f'' : α → E''} {g'' : α → F''} {k'' : α → G''}
+variable {l l' : Filter α}
+
+
+
```

Comment:
```suggestion

```

Reply:
(I'm not completely sure that the diff I created here is the right one, but I hope you understand what my intention was!)